vm/src/vm/vm_ops.rs

```diff
@@ -174,160 +177,348 @@ impl VirtualMachine {
         })
     }
 
-    pub fn _sub(&self, a: &PyObject, b: &PyObject) -> PyResult {
-        self.call_or_reflection(
-            a,
-            b,
-            identifier!(self, __sub__),
-            identifier!(self, __rsub__),
-            |vm, a, b| Err(vm.new_unsupported_binop_error(a, b, "-")),
-        )
+    fn binary_op1<F>(
+        &self,
+        a: &PyObject,
+        b: &PyObject,
+        op_slot: PyNumberMethodsOffset,
+        unsupported: F,
+    ) -> PyResult
+    where
+        F: Fn(&VirtualMachine, &PyObject, &PyObject) -> PyResult,
+    {
+        let num_a = a.to_number();
+        let num_b = b.to_number();
+
+        let slot_a = num_a.methods(&op_slot, self)?.load();
+        let slot_b = num_b.methods(&op_slot, self)?.load();
+
+        if let Some(slot_a) = slot_a {
+            if let Some(slot_b) = slot_b {
+                // Check if `a` is subclass of `b`
+                if b.fast_isinstance(a.class()) {
+                    let ret = slot_b(num_a, b, self)?;
+                    if ret.rich_compare_bool(
```

Comment:
@youknowone 
Is `rich_compare_bool` slow since it calls `_cmp` which is not a trivial function?
Is it the suggested way to check whether `ret` is `self.ctx.not_implemented`?

Reply:
How about using `ret.fast_isinstance(PyNotImplemented::class(self))` to check `self.ctx.not_implemented`?

Reply:
I think this check can be fine:
```rust
ret.is(&self.ctx.not_implemented)
```
But not 100% sure because some type can override operators in weird way and rich_compare_bool will respect it. I have no idea about how CPython handle it.

Reply:
Oh, I missed the second comment. In this case, `fast_isinstance` doesn't have any advantage to simple `is` calling. Because `fast_isinstance` doesn't check `__isinstance__`  python-implemented weird things will not be covered. But because `NotImplemented` is always guaranteed to be singleton, `fast_isinstance` doesn't check more thing. So `fast_isinstance` is not strong as `rich_compare` but same weak as simple `is`.